lib/perl/Genome/SoftwareResult/StageableSimple/SingleFile.pm

```diff
@@ -0,0 +1,28 @@
+package Genome::SoftwareResult::StageableSimple::SingleFile;
+
+use strict;
+use warnings;
+use Genome;
+use File::Basename;
+
+class Genome::SoftwareResult::StageableSimple::SingleFile {
+    is => 'Genome::SoftwareResult::StageableSimple',
+    is_abstract => 1,
+};
+
+sub _temp_staging_file_path {
+    my $self = shift;
+    return File::Spec->join($self->temp_staging_directory, $self->_file_name);
+}
+
+sub _file_name {
+    my $self = shift;
+    $self->fatal_message("Abstract method (_file_name). Needs to be overwritten in your child class");
```

Comment:
`Carp::confess` is handy for cases of programmer error, so it's easier to track down where this is getting called on an incorrectly implemented class.